app/qml/components/MMCircleIcon.qml

```diff
@@ -0,0 +1,47 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+
+Rectangle {
+  id: root
+
+  enum Type {Edit, Checkbox, Close, SmallCheckmark}
+
+  required property int type
+  required property url source
```

Comment:
Source is set internally - this should be hidden to class API - user just select type.